main.go

```diff
@@ -21,7 +21,7 @@ func main() {
 
 	if debugMode {
 		//nolint:staticcheck // SA1019 ignore this!
-		err := plugin.Debug(context.Background(), "registry.terraform.io/opensearch-project/terraform-provider-opensearch",
+		err := plugin.Debug(context.Background(), "registry.terraform.io/opensearch-project/opensearch",
```

Comment:
I think the URL has to be https://registry.terraform.io/providers/opensearch-project/opensearch

Reply:
This is actually unreleated to the fix, so I could leave out this change. 

Nevertheless terraform prints a warning, that providers should not be prefixed with `terraform-provider-` and using `registry.terraform.io/opensearch-project/opensearch` worked in our environment.

Reply:
Hey @moritzzimmer if its unrelated can you please change back to `registry.terraform.io/opensearch-project/terraform-provider-opensearch`? We can track this as a separate issue.